src/client/pool.rs

```diff
@@ -60,6 +62,12 @@ struct Connections<T> {
     // These are internal Conns sitting in the event loop in the KeepAlive
     // state, waiting to receive a new Request to send on the socket.
     idle: HashMap<Key, Vec<Idle<T>>>,
+    // A count of Conns, could be connecting, in-progress, idle, etc.
+    // This is used to keep a maximum bound on the number of connections
+    // This is a WeakCounter so it is not counted towards the Conns count.
+    conns_count: WeakCounter,
+    // A maximum number of idle connections
```

Comment:
is it a maximum number of idle connections, or total connections? I thought the goal was to prevent a massive spike in number of total connections under heavy load

Reply:
Total connections - I think this comment is just old

Reply:
Done